packages/office-ui-fabric-react/src/components/Breadcrumb/Breadcrumb.base.tsx

```diff
@@ -0,0 +1,147 @@
+import * as React from 'react';
+import {
+  BaseComponent,
+  autobind,
+  customizable,
+  getRTL
+} from '../../Utilities';
+import { IFocusZone, FocusZone, FocusZoneDirection } from '../../FocusZone';
+import {
+  IBreadcrumb,
+  IBreadcrumbProps,
+  IBreadcrumbItem,
+  IBreadcrumbStyles,
+  IBreadcrumbStyleProps
+} from './Breadcrumb.props';
+import { DirectionalHint } from '../../common/DirectionalHint';
+import { ResizeGroup } from '../../ResizeGroup';
+import { TooltipHost, TooltipOverflowMode } from '../../Tooltip';
+import { CrumbBase } from './Crumb.base';
+import {
+  classNamesFunction
+} from './utils/index';
+
+export interface IBreadCrumbData {
+  props: IBreadcrumbProps;
+  renderedItems: IBreadcrumbItem[];
+  renderedOverflowItems: IBreadcrumbItem[];
+}
+
+const OVERFLOW_KEY = 'overflow';
+const nullFunction = (): null => null;
+const getClassNames = classNamesFunction<IBreadcrumbStyleProps, IBreadcrumbStyles>();
+
+@customizable('Breadcrumb', ['theme'])
+export class BreadcrumbBase extends BaseComponent<IBreadcrumbProps, any> implements IBreadcrumb {
+  public static defaultProps: IBreadcrumbProps = {
+    items: [],
+    maxDisplayedItems: 999
+  };
+
+  private _classNames: {[key in keyof IBreadcrumbStyles]: string };
+  private _focusZone: IFocusZone;
+
+  constructor(props: IBreadcrumbProps) {
+    super(props);
+  }
+
+  public render() {
+    const {
+      onReduceData = this._onReduceData,
+      maxDisplayedItems,
+      className,
+      theme,
+      items,
+      onRenderItem = this._onRenderItem,
+      getStyles
+    } = this.props;
+
+    const breadCrumbData: IBreadCrumbData = {
+      props: this.props,
+      renderedItems: items.slice(-maxDisplayedItems!),
+      renderedOverflowItems: items.slice(0, -maxDisplayedItems!)
+    };
+
+    this._classNames = getClassNames(getStyles!, { theme: theme!, className });
+
+    return (
+      <ResizeGroup
```

Comment:
Are you setting a cache key on the data to prevent remeasuring during resize? Can one be assigned reliably?